webserver/webserver/src/test/java/io/helidon/webserver/DataChunkReleaseTest.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (c) 2020 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.webserver;
+
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Handler;
+import java.util.logging.Level;
+import java.util.logging.LogRecord;
+import java.util.logging.Logger;
+
+import io.helidon.common.http.Http;
+import io.helidon.media.common.DefaultMediaSupport;
+import io.helidon.webserver.utils.SocketHttpClient;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.endsWith;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.util.ResourceLeakDetector;
+import io.netty.util.internal.StringUtil;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class DataChunkReleaseTest {
+
+    static Logger dataChunkLogger = Logger.getLogger(ByteBufRequestChunk.class.getName());
+    static Logger leakDetectorLogger = Logger.getLogger(io.netty.util.ResourceLeakDetector.class.getName());
+
+    static volatile boolean leakIntercepted = false;
+
+    private static String originalLeakDetectionLevel;
+    private static String originalLeakDetectionSamplingInterval;
+    private static final Handler testHandler = new Handler() {
+        @Override
+        public void publish(final LogRecord record) {
+            if (record.getLevel() == Level.WARNING &&
+                    record.getMessage()
+                            .startsWith("LEAK: RequestChunk.release() was not called before it was garbage collected.")) {
+                leakIntercepted = true;
+            }
+            if (record.getLevel() == Level.SEVERE &&
+                    record.getMessage()
+                            .startsWith("LEAK: ByteBuf.release() was not called before it's garbage-collected.")) {
```

Comment:
"`ByteBuf.release()` was not called before it was garbage collected"

Reply:
This is an assertion for a log message emitted by Netty, see https://github.com/netty/netty/blob/44f85bba5f47df885dbbe5243d008220bfbab5ca/common/src/main/java/io/netty/util/ResourceLeakDetector.java#L320

Reply:
Oh! I see; @danielkec I suppose you should revert the change then.

Reply:
Need for better commentary empirically verified 😄

Reply:
Added better description, the test is more complicated than I would like it to be, I just didn't found any nicer way to do it without heavy mocking.